interfaces/HO-Portal/src/app/programs/program-details/program-details.component.ts

```diff
@@ -1,32 +1,108 @@
 import { Component, OnInit } from '@angular/core';
-import { Router, ActivatedRoute } from '@angular/router';
+import { ActivatedRoute } from '@angular/router';
 import { ProgramsServiceApiService } from 'src/app/services/programs-service-api.service';
 import { Program } from 'src/app/models/program.model';
+import { ModalController } from '@ionic/angular';
+import { ProgramModalPage } from '../program-modal/program-modal.page';
+import { OverlayEventDetail } from '@ionic/core';
+import { TranslateService } from '@ngx-translate/core';
 
 @Component({
   selector: 'app-program-details',
   templateUrl: './program-details.component.html',
   styleUrls: ['./program-details.component.scss'],
 })
 export class ProgramDetailsComponent implements OnInit {
-  public program: any;
+  public languageCode: string;
+  public fallbackLanguageCode: string;
+
+  public program: Program;
+  public programTitle: string;
+  public programArray: any;
+
+  private techFeatures = [
+    'countryId',
+    'schemaId',
+    'credDefId',
+    'credOffer',
+    'proofRequest',
+  ];
+
 
   constructor(
     private route: ActivatedRoute,
-    private router: Router,
-    private programsService: ProgramsServiceApiService
-  ) { }
+    private programsService: ProgramsServiceApiService,
+    public modalController: ModalController,
+    public translate: TranslateService,
+  ) {
+
+  }
 
   ngOnInit() {
+    this.fallbackLanguageCode = this.translate.getDefaultLang();
+    this.languageCode = this.translate.currentLang;
+
     const id = this.route.snapshot.paramMap.get('id');
     this.programsService.getProgramById(id).subscribe((response) => {
-      this.program = this.generateArray(response);
-      console.log(this.program);
+      this.program = response;
+      this.programTitle = this.mapLabelByLanguageCode(this.program.title);
+      this.programArray = this.generateArray(this.program);
     });
   }
 
+
+
   public generateArray(obj) {
-    return Object.keys(obj).map((key) => ({ key, value: obj[key] }));
+    const result = [];
+    for (const key in obj) {
+      if (this.techFeatures.indexOf(key) <= -1) {
+
+        const keyNew = this.translate.instant('page.programs.program-details.' + key);
+        const valueNew = this.mapLabelByLanguageCode(obj[key]);
+
+        let isArray = false;
+        if (valueNew instanceof Array) {
+          if (typeof valueNew[0] === 'object') {
+            isArray = true;
+            // Enter code here to visualize array-properties (like Criteria/Aidworkers) differently
+          }
+        }
+        result.push(({ key: keyNew, value: valueNew, isArray }));
+      }
+    }
+    return result;
+  }
+
+  private mapLabelByLanguageCode(property: any) {
+    let label = property[this.languageCode];
+
+    if (!label) {
+      label = property[this.fallbackLanguageCode];
+    }
+
+    if (!label) {
+      label = property;
+    }
+
+    return label;
+  }
+
+  async openModal() {
```

Comment:
I would advice to name things not _too generic_...
As this will have some specific function or goal, so `showProgramProperties` or `openProgramJson` would be better. Because why would this method-name need to change if we want to open the details/json in a different component?

That way names-of-things will be long, bug not confusing. For example "ProgramModalPage": Is it a page? Is it a Modal? 🤷‍♂  -> It is a "thing that displays program-properties".

Reply:
Point taken. Changed openModal fn-name. (I took this directly from an example and didn't change it.) And changed name of programModalPage.